lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -598,54 +598,43 @@ function writeStringsToCtcFiles(locale, strings) {
 }
 
 /**
- * This function does three things:
+ * This function does two things:
  *
- *    - Add `meaning` property to ctc messages that have the same message so TC can disambiguate (otherwise it fails to import).
- *    - Throw if the `meaning` of any collisions *also* collides (can't disambiguate messages).
+ *    - Add `meaning` property to ctc messages that have the same message but different descriptions so TC can disambiguate.
  *    - Throw if the known collisions has changed at all.
  *
  * @param {Record<string, CtcMessage>} strings
  */
 function resolveMessageCollisions(strings) {
-  /** @type {Map<string, Array<[string, CtcMessage]>>} */
+  /** @type {Map<string, Array<CtcMessage>>} */
   const stringsByMessage = new Map();
 
   // Group all the strings by their message.
-  for (const entry of Object.entries(strings)) {
-    const collisions = stringsByMessage.get(entry[1].message) || [];
-    collisions.push(entry);
-    stringsByMessage.set(entry[1].message, collisions);
+  for (const ctc of Object.values(strings)) {
+    const collisions = stringsByMessage.get(ctc.message) || [];
+    collisions.push(ctc);
+    stringsByMessage.set(ctc.message, collisions);
   }
 
-  /** @type {Array<[string, CtcMessage]>} */
+  /** @type {Array<CtcMessage>} */
   const allCollisions = [];
-  for (const group of stringsByMessage.values()) {
+  for (const messageGroup of stringsByMessage.values()) {
     // If this message didn't collide with anything else, skip it.
-    if (group.length <= 1) continue;
-    allCollisions.push(...group);
+    if (messageGroup.length <= 1) continue;
 
-    // We have a message collision, time to check collisions on the `meaning` property.
-    /** @type {Map<string|undefined, Array<[string, CtcMessage]>>} */
-    const stringsByMeaning = new Map();
-    for (const [key, ctc] of group) {
-      ctc.meaning = ctc.description;
-
-      const collisions = stringsByMeaning.get(ctc.meaning) || [];
-      collisions.push([key, ctc]);
-      stringsByMeaning.set(ctc.meaning, collisions);
-    }
+    // If group shares both message and description, they can be translated as if a single string.
+    const descriptions = new Set(messageGroup.map(ctc => ctc.description));
+    if (descriptions.size <= 1) continue;
 
-    for (const meaningGroup of stringsByMeaning.values()) {
-      if (meaningGroup.length <= 1) continue;
-
-      const debugMeaningList = meaningGroup.map(entry => [entry[0], entry[1].meaning].join('\n'));
-      const debugCollisionsMessage = `${meaningGroup[0][1].message}\n\n${debugMeaningList.join('\n\n')}`;
-      throw new Error(`Each strings' \`message\` or \`description\` must be different for the translation pipeline. The following keys did not have unique \`meaning\` values:\n\n${debugCollisionsMessage}`);
+    // We have duplicate messages with different descriptions. Disambiguate using `meaning` for TC.
+    for (const ctc of messageGroup) {
+      ctc.meaning = ctc.description;
```

Comment:
There's a few ways to do this. For instance, if this `messageGroup` had three strings with one `description` and two strings with a different `description`, the three-string set could get no `meaning` while the two-string set did get a `meaning`.

However since this is currently a hypothetical situation and `meaning` is hashed into the ID (not explicitly included in the messages), it doesn't matter a whole lot and assigning a meaning to any `messageGroup` with multiple `description`s is a lot simpler.